source/runbooks/revoke-network-access.html.md.erb

```diff
@@ -0,0 +1,96 @@
+
+## Stopping access to the internet for an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+The process needs to be done for all 4 regions listed. An example is
+
+- hmpps-production-data-nacl
+- hmpps-production-private-nacl
+- hmpps-production-protected-nacl
+- hmpps-production-public-nacl
+
+Follow the process below for each of these
+
+1. Go to VPC->Security->Network ACLs
+2. Click on the checkbox for the one you want to change then select **outbound rules**.
+3. Due to the way checks work in AWS (see [here](https://docs.aws.amazon.com/network-firewall/latest/developerguide/suricata-rule-evaluation-order.html) ) we need to amend each one in place to Deny where it is currently Allow. **Note** any that are currently set to Deny in case this is manually changed back.
+4. Save changes
+
+Setting up a deny rule on the inline-inspection firewalls, regardless of the processing order, is not possible due to the issue mentioned above. This is why the deny on each area is required.
+
+## Stopping access from the MOJ to an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+The process needs to be done for all 4 regions listed. An example is
+
+- hmpps-production-data-nacl
+- hmpps-production-private-nacl
+- hmpps-production-protected-nacl
+- hmpps-production-public-nacl
+
+You need to access this through the same route as above:
+
+1. Go to VPC->Security->Network ACLs
+2. Click on the checkbox for the one you want to change then select **inbound rules**.
+3. To prevent access change the one for the current system to Deny. For example in production you would amend 10.27.8.0/21 to deny for hmpps-production.
+4. Ensure that the access to the system is completely denied by amending the TCP to ports 1024 - 65535 on 0.0.0.0/0 to Deny too.
+5. Save the changes
+
+As mentioned above the way checks are undertaken in AWS (see [here](https://docs.aws.amazon.com/network-firewall/latest/developerguide/suricata-rule-evaluation-order.html) ) mean adding a step for Deny may not work.
+
+## Stopping access to the MOJ for an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+The process needs to be done for all 4 regions listed. An example is
+
+- hmpps-production-data-nacl
+- hmpps-production-private-nacl
+- hmpps-production-protected-nacl
+- hmpps-production-public-nacl
+
+You need to access this through the same route as above:
+
+1. Go to VPC->Security->Network ACLs
+2. Click on the checkbox for the one you want to change then select **outbound rules**
+3. To prevent access change the one for the current system to Deny. For example in production you would amend 10.27.8.0/21 to deny for hmpps-production.
+4. Ensure that the access to the system is completely denied by amending the TCP to ports 1024 - 65535 on 0.0.0.0/0 to Deny too.
+5. Save the changes
+
```

Comment:
The documentation so far addresses (although in some cases it is not correct)

-  Blocking access for something entering via the internet
-  Blocking access for something leaving via the internet
-  Blocking access for something leaving via an internal route (eg, TGW)
-  Blocking access for something entering via an internal route (again, eg TGW)

 
 but it is still missing:

-   Blocking access for something moving internally inside the Modernisation Platform

  and ignores looking at the problem from the other angle:
   - isolating a single service/app or an environment (or even worth considering isolating a business unit as oppose to as single service/app)
   -  blocking ingress/egress in multiple or all services/environments

Reply:
I may have covered this with my other changes. They will be posted later.

Reply:
Regarding isolating a single business unit, that ought to be covered by the guidance on detaching an internet gateway, and putting a deny rule at the top of the ingress/egress NACL rules. In theory you could also do this by removing the TGW route table association in `core-network-services` but that would only cover part of the routing (enough to break TCP sessions, but not to stop UDP traffic coming in).

Reply:
I have added additional code to list what to do to remove the transit gateway. This will prevent an environment routing around through others.